### DISCUSSION TITLE
Update all packages

### DISCUSSION THREAD

**Discussion by o-mingya** [winpython/winpython#1093] [Q&A] [answered]
Is there a script to update all packages, similar to the upgrade_pip.bat but for all outdated packages?

**Comment by stonebig** [comment, marked as answer]
Getting a "set" of package that roughly work together is tricky, and takes time:
- determine what package has a more recent version,
- try to upgrade it,
- discover it can't because another package has flagged it as "not compatible", or worse, as not yet flagged it as not compatible while it is indeed not,
- finding a middle way, sometimes fixing one yourself upstream, ....

What would work in theory, would be a published a "pip freeze" of Winpython betas (or last official), that you may use as a pip upgrade list, if all packages come "as is" from "pip official site".

**Reply by o-mingya** [reply]
Would pip-review work the same as pip freeze?

**Reply by stonebig** [reply]
not the same thing.

**Reply by o-mingya** [reply]
I tried the pip freeze upgrade list. I'm encountering conflicting requirements with h2, hyper versions. The unfortunate thing is, it happens only on amd64 version, but not on 32bit. Even when I have both distribution versions available, it kept pointing to the lowest one. Any pointer on how to resolve this?
